pages/_app.tsx

```diff
@@ -1,18 +1,56 @@
 import { ChakraProvider } from '@chakra-ui/react'
 import { Fonts, Navbar, theme } from 'components/Layout'
 import { AppProps } from 'next/app'
+import { useRouter } from 'next/router'
+import Script from 'next/script'
+import { useEffect } from 'react'
 import { QueryClient, QueryClientProvider } from 'react-query'
+import { sendPageView } from '../gtag'
 
 const queryClient = new QueryClient()
 
 export default function App({ Component, pageProps }: AppProps): JSX.Element {
+  // send URL changes to Google Analytics
+  const router = useRouter()
+  useEffect(() => {
+    const handleRouteChange = (url: string) => {
+      sendPageView(url)
+    }
+
+    router.events.on('routeChangeComplete', handleRouteChange)
+    router.events.on('hashChangeComplete', handleRouteChange)
+
+    // unsubscribe from events on unmount
+    return () => {
+      router.events.off('routeChangeComplete', handleRouteChange)
+      router.events.off('hashChangeComplete', handleRouteChange)
+    }
+  }, [router.events])
```

Comment:
According to [this article](https://medium.com/@mikegajdos81/how-to-add-googleanalytics-4-to-nextjs-app-in-4-simple-steps-3c6f9de2f866), since Next.js generates a single page app (because it's a React framework), Google Analytics won't track route changes. This is why we need to set that up ourselves in this `useEffect()`.